cmd/nexusUpload.go

```diff
@@ -0,0 +1,62 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	piperHttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+)
+
+type artifactDescription struct {
+	ID         string `json:"id"`
+	Classifier string `json:"classifier"`
+	Type       string `json:"type"`
+	File       string `json:"file"`
+}
+
+func nexusUpload(config nexusUploadOptions, telemetryData *telemetry.CustomData) error {
+	log.Entry().Info(config)
+
+	log.Entry().Info("JSON string is ", config.Artifacts)
+
+	var artifacts []artifactDescription
+	err := json.Unmarshal([]byte(config.Artifacts), &artifacts)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to convert JSON: %s", err))
```

Comment:
We should not panic here.
Rather use `log.Entry().WithError(err).Fatal()`

With that we can make use of the exit handler which is defined in the generated part.
This helps us to e.g. send telemetry data (including the info that an error occured) and perform other actions we defined any may define in the future.

Reply:
sounds to me like this should be part of static code checks

Reply:
Why not letting errors bubble up normally until they reach the corba framework (generated code). Otherwise we end up in a lot of cases where we randomly exit.

Beside that I doubt it is the duty of a logging framework to perform exits, but that is another story.